### ISSUE TITLE
Webpack 5 HMR Issue: Cannot set property './index.js' of undefined

### ISSUE THREAD

**Issue by smparsons** [webpack/webpack#10790]
# Bug report

**What is the current behavior?**

In Webpack 5, when I am using webpack-hot-middleware and `HotModuleReplacementPlugin`, then if I make a change in one of my JavaScript files and refresh my web app, I receive the error `Uncaught TypeError: Cannot set property './index.js' of undefined` in the file `jsonp_chunk_loading`. 

**If the current behavior is a bug, please provide the steps to reproduce.**

I created a minimal repo to reproduce the issue here: https://github.com/smparsons/webpack-hmr-issue. Steps to reproduce are in the README file.

**What is the expected behavior?**

I would expect to not receive the above error after making a change in my code and refreshing my web app.

**Other relevant information:**
webpack version:  5.0.0-beta.15
Node.js version: 11.10.1
Operating System: macOS Catalina Version 10.15
Additional tools:

**Comment by chenxsan** [comment]
Looks like same problem I have:

![image](https://user-images.githubusercontent.com/1091472/80433625-04c6f180-892a-11ea-9f4c-57695b514396.png)

I've left a note here https://github.com/webpack/webpack/issues/10534 waiting for feedback.

**Comment by sokra** [comment]
@jantimon HTML looks like:

``` html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>Test</title>
<script src="scripts/main.2e4e6e36937f0a3989bd.js"></script><script src="main.abf4d034b3614387dfd4.hot-update.js"></script></head>
<body>
    <div id="content"></div>
</body>
</html>
```

`<script src="main.abf4d034b3614387dfd4.hot-update.js"></script>` should not be there. Maybe filter out assets that have `hotModuleReplacement` or `development` flagged.

---

I was able to fix it by adding these lines at https://github.com/jantimon/html-webpack-plugin/blob/96e0ae607afa7295c40d0a5073c791189def6337/index.js#L581-L583

``` diff
       const entryPointPublicPaths = entryPointFiles
+        .filter(chunkFile => {
+          const info = compilation.getAsset && compilation.getAsset(chunkFile).info;
+          return !(info && (info.hotModuleReplacement || info.development));
+        })
         .map(chunkFile => {
           const entryPointPublicPath = publicPath + this.urlencodePath(chunkFile);
```

**Comment by jantimon** [comment]
Hey @sokra thanks for analysing the problem 👍

The same issue was also reported here: https://github.com/jantimon/html-webpack-plugin/issues/1415

Is `compilation.getAsset` new for webpack 5 or why do you need `compilation.getAsset &&`?  

I'll try to add that fix asap 👍

**Comment by sokra** [comment]
Since 4.40 [Commit](https://github.com/webpack/webpack/commit/758269e81456c946a96b521ee936dbec99d07132)

As fallback maybe check for `/\.hot-update\.js$/` or don't check or set min. webpack version.

**Comment by jantimon** [comment]
Thank you so much - I'll let you know once this is released 👍

**Comment by smparsons** [comment]
Thanks guys!

**Comment by jantimon** [comment]
Released as html-webpack-plugin@4.2.1

**Comment by evilebottnawi** [comment]
@jantimon Thanks!

Let's close the issue

@smparsons feel free to leave feedback or open a new issue if you still have problems
